app/App.hs

```diff
@@ -77,10 +80,29 @@ runAppIO args@RunAppIOArgs {..} =
     ExitJuvixError e -> do
       printErr e
       embed exitFailure
-    ExitMsg exitCode t -> embed (putStrLn t >> hFlush stdout >> exitWith exitCode)
+    ExitMsg exitCode t -> exitMsg' exitCode t
     SayRaw b -> embed (ByteString.putStr b)
   where
+    exitMsg' :: ExitCode -> Text -> Sem r x
+    exitMsg' exitCode t = embed (putStrLn t >> hFlush stdout >> exitWith exitCode)
+    getMainFile' :: Maybe (AppPath File) -> Sem r (Path Abs File)
+    getMainFile' m = case m of
+      Just p -> embed (prepathToAbsFile invDir (p ^. pathPath))
+      Nothing -> case pkg ^. packageMain of
+        -- TODO do something special if it is the global package?
```

Comment:
What do you mean by "special"?

Reply:
I was thinking that maybe we want to explicitly forbid the `main` file in the global project.
I don't think there is ever a usecase for running `juvix compile` outside of a project.
By default, the global project doesn't have a `main` file by default, so maybe we leave it as it is and remove the `TODO` comment